catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -146,28 +215,49 @@ public Metacard transform(InputStream inputStream)
     @Override
     public Metacard transform(InputStream inputStream, String id)
             throws IOException, CatalogTransformerException {
+        return handleTransform(inputStream, id);
+    }
 
-        String xml = null;
-        XstreamPathValueTracker pathValueTracker = null;
-        xml = IOUtils.toString(inputStream);
+    private Metacard handleTransform(InputStream inputStream, String id)
+            throws CatalogTransformerException {
+        String xml;
+        XstreamPathValueTracker pathValueTracker;
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+            readInputStreamIntoByteArrayOutputStream(inputStream, baos);
```

Comment:
We should really never be reading any externally obtained input stream to a byte array. What if someone passes in a XXGB stream? If you can guarantee the length, that might be different. However, in the case where it is ok to go InputStream -> Byte Array, then Joe is absolutely correct, use IOUtils.